jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/core/ls/ArgumentUtils.java

```diff
@@ -139,20 +150,32 @@ public static Map<String, Object> getObject(Map<String, Object> obj, String key)
     }
 
     /**
-     * Returns the child as a JsonObject if it exists and is an object, and null
-     * otherwise
+     * Returns the child as a JSON array if the data parameter contains an array of strings; otherwise, 
+     * returns it as an object if present, or null if not.
      *
-     * @param obj the object to get the child of
+     *
+     * @param data the object to get the child of
      * @param key the key of the child
-     * @return the child as a JsonObject if it exists and is an object, and null
-     *         otherwise
+     * @return Returns the child as a JSON array if the data parameter contains an array of strings; otherwise, 
+     * returns it as an object if present, or null if not.
      */
-    public static JsonObject getObjectAsJson(Map<String, Object> obj, String key) {
-        Object child = obj.get(key);
-        if (child != null && child instanceof Map<?, ?>) {
+    public static Object getValueFromDataParameter(Map<String, Object> data, String key) {
+
+        Object child = data.get(key);
+        if (child != null && child instanceof String) {
+        	// if the value in the 'data' is a string, we string the object. 
+        	// eg: data = “AssertTrue”
+            return child;
+        } else if (child instanceof List<?>) {
+        	// if the value in the 'data' is an array, we will convert it to an JsonArray.
+        	// eg: data =[“ApplicationScoped", "RequestScoped”]
             Gson gson = new Gson();
-            return (JsonObject) gson.toJsonTree(child);
+            JsonArray jsonArray = gson.toJsonTree(child).getAsJsonArray();
+            return jsonArray;
+        } else {
+        	//Returns the object if it exists and is an object, and null otherwise
+            return getObject(data, key);
```

Comment:
Could you clarify why we need this line? It checks if the data is a Map?

Reply:
I was expecting the 'getObject' method to return an object, as its return type is 'Map<String, Object>. Could I replace it with 'return null'?

Reply:
It would probably return null all the time anyway, thanks.